### PR TITLE
Update upload artifact and add option for extra repositories

### PR DESCRIPTION
**Pull request by seabbs**
As the title, this PR updates the version of upload artifacts which was throwing a depreciation error for me. 

I have also added a pass-through of the `extra_repositories` option for `setup-r` as our use case needs this in order to install our dependencies (I can split this off into another PR if desired).

I have this running here from my fork and it looks like everything is now working as expected: https://github.com/epinowcast/primarycensoreddist/actions/runs/10906920699?pr=79

The non-forked version is failing here with the depreciation note I mentioned: https://github.com/epinowcast/epinowcast/actions/runs/10839748687?pr=528

Thanks a lot for keeping this ticking over. We find it very useful for a lot of our work.

### CODE REVIEW

**Comment by lorenzwalthert** [comment]
Thanks, any objections @assignUser and @plietar?

**Comment by seabbs** [comment]
To get this fully working I had to switch out to use `download-artifact` in the comment action. Can see this working here: https://github.com/epinowcast/primarycensoreddist/pull/80

File: actions/comment/action.yaml [hunk 1]
```diff
@@ -9,27 +9,12 @@ runs:
...
-          fs.writeFileSync('${{github.workspace}}/pr.zip', Buffer.from(download.data));
-    - run: unzip pr.zip
-      shell: bash
+        name: pr
+        github-token: ${{ inputs.GITHUB_TOKEN }}
+        repository: ${{ github.repository }}
+        run-id: ${{github.event.workflow_run.id }}
     - name: 'Comment on PR'
       id: 'comment'
       uses: actions/github-script@v3
```
Comment (assignUser) [inline actions/comment/action.yaml:20]: we should also bump this, latest is 7... 3 is probably already deprecated?

Comment (sbfnk) [reply actions/comment/action.yaml:20]: done in https://github.com/lorenzwalthert/touchstone/pull/144

Comment (lorenzwalthert) [reply actions/comment/action.yaml:20]: @assignUser maybe we can automate that with dependabot?

**Comment by seabbs** [comment]
I’ve updated to include the changes from @sbfnk in #144

**Comment by seabbs** [comment]
I see only this with write access to this repository can merge this or

**Comment by lorenzwalthert** [comment]
Oh ok 😄.

**Pull request merged**